tools/arcilator/arcilator.cpp

```diff
@@ -333,6 +346,32 @@ static LogicalResult processBuffer(
   if (!module)
     return failure();
 
+#ifdef ARCILATOR_ENABLE_JIT
+  // Check JIT execution is valid.
+  if (!runSimulation.empty()) {
+    Operation *toCall = module->lookupSymbol(runSimulation);
+    if (!toCall) {
+      llvm::errs() << "entry point not found: '" << runSimulation << "'\n";
+      return failure();
+    }
+
+    if (!llvm::isa<LLVM::LLVMFuncOp, func::FuncOp>(toCall)) {
+      llvm::errs() << "entry point '" << runSimulation
+                   << "' was found but on an operation of type '"
+                   << toCall->getName() << "' while a function was expected\n";
+      llvm::errs() << "supported functions: 'func.func', 'llvm.func'\n";
+      return failure();
+    }
+
+    auto toCallFunc = llvm::cast<FunctionOpInterface>(toCall);
+    if (toCallFunc.getNumArguments() != 0) {
+      llvm::errs() << "entry point '" << runSimulation
+                   << "' must have no arguments\n";
+      return failure();
+    }
+  }
+#endif // ARCILATOR_ENABLE_JIT
```

Comment:
What's the consideration of checking this before the lowering rather than once everything is lowered to LLVM. At that point you wouldn't have to check for `func.func` and could also support other custom functions, right?

Reply:
I am sure I had a reason to do that but I cannot remember it… I guess I’ll move it down, perhaps while doing so it will come back to me.

Reply:
I remember now! It was so the output could contain the name of the operation it was pointing to so that it is consistent with the input. It would be weird for the error message to talk about a lowered state of the op, one could end up being very confused. Do you think this is worth it?

Reply:
It's nice to have good error messages, but I think here it's not worth it.